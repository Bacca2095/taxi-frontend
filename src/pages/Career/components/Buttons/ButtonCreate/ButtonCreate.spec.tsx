import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ButtonCreate } from '.';

describe('ButtonCreate tests', () => {
  let onClick: jest.Mock;

  beforeEach(() => {
    onClick = jest.fn();
  });

  it('should match snapshot', () => {
    const { container } = render(<ButtonCreate onClick={onClick} />);

    expect(container).toMatchSnapshot();
  });

  it('should click on create', async () => {
    const { findByTestId } = render(<ButtonCreate onClick={onClick} />);

    const createButton = await findByTestId('create');

    fireEvent.click(createButton);

    expect(onClick).toBeCalled();
  });
});
