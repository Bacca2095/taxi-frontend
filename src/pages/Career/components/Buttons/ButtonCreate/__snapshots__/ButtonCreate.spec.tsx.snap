// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCreate tests should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
    data-testid="create"
    tabindex="0"
    title="Crear Carrera"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm4 11h-3v3c0 .55-.45 1-1 1s-1-.45-1-1v-3H8c-.55 0-1-.45-1-1s.45-1 1-1h3V8c0-.55.45-1 1-1s1 .45 1 1v3h3c.55 0 1 .45 1 1s-.45 1-1 1z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
